src/Microsoft.Health.Dicom.Core/Features/ChangeFeed/ChangeFeedHandler.cs

```diff
@@ -8,24 +8,33 @@
 using System.Threading.Tasks;
 using EnsureThat;
 using MediatR;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Features.Common;
+using Microsoft.Health.Dicom.Core.Features.Security;
+using Microsoft.Health.Dicom.Core.Features.Security.Authorization;
 using Microsoft.Health.Dicom.Core.Messages.ChangeFeed;
 
 namespace Microsoft.Health.Dicom.Core.Features.ChangeFeed
 {
-    public class ChangeFeedHandler : IRequestHandler<ChangeFeedRequest, ChangeFeedResponse>
+    public class ChangeFeedHandler : BaseHandler, IRequestHandler<ChangeFeedRequest, ChangeFeedResponse>
     {
         private readonly IChangeFeedService _changeFeedService;
 
-        public ChangeFeedHandler(IChangeFeedService changeFeedService)
+        public ChangeFeedHandler(IDicomAuthorizationService dicomAuthorizationService, IChangeFeedService changeFeedService)
+            : base(dicomAuthorizationService)
         {
-            EnsureArg.IsNotNull(changeFeedService, nameof(changeFeedService));
-
-            _changeFeedService = changeFeedService;
+            _changeFeedService = EnsureArg.IsNotNull(changeFeedService, nameof(changeFeedService));
         }
 
         public async Task<ChangeFeedResponse> Handle(ChangeFeedRequest request, CancellationToken cancellationToken)
         {
             EnsureArg.IsNotNull(request, nameof(request));
+
+            if (await AuthorizationService.CheckAccess(DataActions.Read, cancellationToken) != DataActions.Read)
```

Comment:
This signature is interesting, I was expecting
bool CheckAccess(ExpectedActions), instead we pass and return Actions and then compare here, why?

Reply:
We're following the FHIR implementation for this currently. The reasoning for FHIR was to allow the following scenarios (maybe more in the future):
- Allow for the possibility to have a downgraded experience based on the permissions
- Allow a single call to get all possible permissions when in the context of a batch/transaction